pkg/il/evaluator/evaluator.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package evaluator
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/hashicorp/golang-lru"
+
+	pb "istio.io/api/mixer/v1/config/descriptor"
+	"istio.io/mixer/pkg/attribute"
+	"istio.io/mixer/pkg/config"
+	"istio.io/mixer/pkg/config/descriptor"
+	"istio.io/mixer/pkg/expr"
+	"istio.io/mixer/pkg/il/compiler"
+	"istio.io/mixer/pkg/il/interpreter"
+)
+
+// ILEvaluator is an implementation of expr.Evaluator that also exposes specific methods.
+// Specifically, it can listen to config change events.
+type ILEvaluator struct {
```

Comment:
We're in package `evaluator`, naming this struct `IL` doesn't lose anything since you refer to it as `evaluator.IL` (vs the redundant `evaluator.ILEvaluator`).